src/core/math/mat4.js

```diff
@@ -778,6 +778,42 @@ class Mat4 {
         return this;
     }
 
+    /**
+     * Sets the matrix to a reflection matrix, which can be used as a mirror transformation by the
+     * plane.
+     *
+     * @param {import('./plane.js').Plane} plane - The plane to reflect by.
+     * @returns {Mat4} Self for chaining.
+     */
+    setReflection(plane) {
```

Comment:
I'm a bit late to the party here, but this is introducing a dependency of the math API on the shape API. I'm not a big fan of that.

Reply:
Good point.
I feel like moving Plane to Math. I even assumed it's there.